### ISSUE TITLE
Feature request: Motion detection

### ISSUE THREAD

**Issue by jat255** [BobStrogg/teslacam-browser#3]
I'm wondering if it would be possible to do something like shown in [this post](https://hackernoon.com/motion-detection-in-javascript-2614adea9325) to analyze each video, and display how much motion (and where in the video) it takes place. This could really help users to key in on where the activity that triggered Sentry Mode took place. I'd envision the scrub bar could colored to show where the motion is (transparent where there's no motion and maybe increasing opacity of blue or something where there is). 

This would probably take some time to process, so maybe it'd be a button to click to process.

**Comment by jat255** [comment]
some other implementations:

- https://codersblock.com/blog/motion-detection-with-javascript/
- https://github.com/jasonmayes/JS-Motion-Detection
- https://trackingjs.com/

Maybe even just grabbing the frames and diffing them to get a number would be good enough...

**Comment by BobStrogg** [comment]
Yep manually scrubbing videos is still super tedious and this would be a great feature.  Have you tried any of these libraries before?  I assume it's something we'd want to kick off as a background task.

**Comment by Breachfront** [comment]
This!  This is what this app needs.  I would love to be able to scroll down and view the streams on a single screen for each day, with a graph of motion displayed at the bottom, under each one.  Makes it easy to scrub through for highlights / triggers.  Wonder if it's possible to play multiple clips at the same time without killing performance?  Having speed modifications (2x, 4x, 8x. etc..) would make a huge difference for reviewing footage without having to unreliably scrub (and possibly miss something).

Love the work so far, tempted to start learning more Javascript to contribute.  I'm a Python man.

**Comment by BobStrogg** [comment]
The latest version has a nice “combined view” that puts all the clips together under one scrub bar, but I love the idea of an activity graph... going to look into that 🤓

**Comment by Breachfront** [comment]
I’ll have to try out the combined view. The last release I had, had only one set of clips viewable at a time and you had to switch between them.

I wonder if we could overlay multiple clips and show the differences on the videos with time stamps on them. They use some software on “Portals to Hell”, a paranormal investigation show, that does exactly that, and it makes it easy to review multiple clips and have the motion events presented to you all at once.
